api/resources/views.py

```diff
@@ -4,18 +4,50 @@
 from rest_framework.exceptions import NotFound
 
 from api.base import permissions as base_permissions
+from api.base.exceptions import JSONAPIException
 from api.base.parsers import (
     JSONAPIMultipleRelationshipsParser,
     JSONAPIMultipleRelationshipsParserForRegularJSON,
 )
 from api.base.views import JSONAPIBaseView
-from api.resources.permissions import ResourceDetailPermission
+from api.resources.permissions import ResourceDetailPermission, ResourceListPermission
 from api.resources.serializers import ResourceSerializer
 from framework.auth.oauth_scopes import CoreScopes
-from osf.models import Guid, OutcomeArtifact
+from osf.models import Guid, OutcomeArtifact, Registration
 
 logger = logging.getLogger(__name__)
 
+
+class ResourceList(JSONAPIBaseView, generics.ListCreateAPIView):
+
+    permission_classes = (
+        ResourceListPermission,
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.READ_REGISTRATION_RESOURCES]
+    required_write_scopes = [CoreScopes.WRITE_REGISTRATION_RESOURCES]
+
+    view_category = 'resources'
+    view_name = 'resource-list'
+
+    serializer_class = ResourceSerializer
+
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON)
+
+    def get_permissions_proxy(self):
+        try:
+            registration_guid = self.request.data['registration']
+        except KeyError:
+            raise JSONAPIException('Must provide "registration" relationship in payload"')
```

Comment:
```suggestion
            raise JSONAPIException('Must provide "registration" relationship in payload')
```

Reply:
Also, it's a good idea to include a `source` field with the JSONAPIException so we can see what field is problematic without reading the error string. Not that this error is likely to come up runtime after development is complete, but still, it's a good practice.